### PR TITLE
New arguments parser

### PR DESCRIPTION
**Pull request by filfat**
There's still a lot to do before this is even close to mergeable.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -143,6 +143,7 @@
     "dependencies": {
         "@beenotung/level-ts": "^1.14.1",
         "@jsprismarine/bedrock-data": "1.16.100",
+        "@jsprismarine/brigadier": "git+https://github.com/JSPrismarine/brigadier.git",
```
Comment (filfat) [inline package.json:146]: Replace this with packaged version.

File: src/Server.ts [hunk 2]
```diff
@@ -375,15 +377,14 @@ export default class Server {
...
-                    .getUsername()
-                    .toLowerCase()
-                    .startsWith(name.toLowerCase())
-            ) ?? null
+    public getPlayerByName(name: string): Player {
+        const player = Array.from(this.players.values()).find((player) =>
+            player.getUsername().toLowerCase().startsWith(name.toLowerCase())
         );
+
+        if (!player) throw new Error(`Can't find player ${name}`);
```
Comment (filfat) [inline src/Server.ts:385]: Same should be done for `getPlayerByExactName`.

File: src/command/CommandManager.ts [hunk 3]
```diff
@@ -92,110 +99,89 @@ export default class CommandManager {
...
-                    .getLogger()
-                    .silly(err.stack, 'CommandManager/dispatchCommand');
-                return;
-            }
+            });
+        } catch (err) {
+            sender.sendMessage(`§c${err}`);
         }
-
-        sender.sendMessage(
```
Comment (filfat) [inline src/command/CommandManager.ts:190]: This functionality should be re-added.

File: src/command/vanilla/BanCommand.ts [hunk 4]
```diff
@@ -11,46 +14,29 @@ export default class BanCommand extends Command {
...
-                }!`
-            );
-
-        return `Banned ${args[0] || sender.getUsername()} ${
-            args.length > 1 ? `for reason ${args.slice(1).join(' ')}` : ''
-        }`;
+    public async register(dispatcher: CommandDispatcher<any>) {
+        dispatcher.register(
+            literal('ban').then(
+                argument('player', string()).executes(async (context) => {
```
Comment (filfat) [inline src/command/vanilla/BanCommand.ts:23]: Add `reason` argument.

File: src/command/vanilla/KickCommand.ts [hunk 5]
```diff
@@ -9,44 +11,25 @@ export default class KickCommand extends Command {
...
-            : 'No reason specified.';
-        const target = sender.getServer().getPlayerByName(`${args[0]}`);
-
-        if (!target) {
-            return sender.sendMessage("§cCan't find the selected player.");
-        }
+    public async register(dispatcher: CommandDispatcher<any>) {
+        dispatcher.register(
+            literal('op').then(
+                argument('player', new CommandArgumentEntity()).executes(
```
Comment (filfat) [inline src/command/vanilla/KickCommand.ts:19]: Add `reason` argument.

**Comment by Bavfalcon9** [comment]
I know this is already merged, but I want to give my opinion on this, 
While there are no errors or anything I see that can be improved upon in this implementation, I believe that this parsing implementation is more complex than it needs to be. 
For example, I believe something like: 
```ts
class Command {
     public constructor() {
          this.arguments.set(0, new Argument('player', 'online player', (argStr) => {
               /* get server player */
          }));
     }
     
     public execute(sender, ...args) {
          const player: Player = args[0]; // absolute or null
     }
}
```
Is more user friendly and potentially more optimized than this current implementation.

**Pull request merged**